controllers/database_reconciler_helpers.go

```diff
@@ -157,7 +157,7 @@ func (r *DatabaseReconciler) handleSync(ctx context.Context, database *ndbv1alph
 	databaseStatus.Type = database.Spec.Instance.Type
 
 	// Provision the database if it has not been provisioned earlier
-	if databaseStatus.Status == "" || databaseStatus.Id == "" {
+	if databaseStatus.Status == "" && databaseStatus.Id == "" {
```

Comment:
I find it confusing that when the database.Status.DeregistrationOperationId is empty we deregister the database from NDB. But when status is empty it means we provision. What do you think?

Reply:
The idea is - We store the operationIds of the operations that we perform on NDB (Provisioning and De-Provisioning) here in our case. 
So if an id is set, it means that the operation was submitted to NDB previously and we do not need to submit the operation. On the other hand, if the id is not set then we're supposed to perform the operation submission based on the current status of the database custom resource.